src/all/novelfull/src/eu/kanade/tachiyomi/extension/all/novelfull/NovelFullGlobal.kt

```diff
@@ -0,0 +1,241 @@
+package eu.kanade.tachiyomi.extension.all.novelfull
+
+import android.util.Log
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.asObservableSuccess
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import org.jsoup.select.Evaluator
+import rx.Observable
+
+abstract class NovelFullGlobal(
+    override val lang: String,
+    val supportsSearch: Boolean = true,
+) : ParsedHttpSource() {
+
+    override val baseUrl = "https://allwebnovel.com" // https://mangafre.com
+
+    override val name = "NovelFull"
+
+    override val supportsLatest = true
+
+    override val client = network.cloudflareClient.newBuilder().build()
+
+    // ============================== Popular ===============================
+
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/comic/bookclass.html?type=hot_novel&page_num=$page&language=$lang")
+    }
+
+    override fun popularMangaSelector(): String {
+        return "#list-page > .col-truyen-main .row"
+    }
+
+    override fun popularMangaFromElement(element: Element): SManga {
+        return SManga.create().apply {
+            title = element.select(Evaluator.Tag("a")).attr("alt")
+            thumbnail_url = element.selectFirst(Evaluator.Tag("img"))!!.imgAttr()
+            setUrlWithoutDomain(element.select(Evaluator.Tag("a")).attr("href"))
```

Comment:
```suggestion
            setUrlWithoutDomain(element.selectFirst(Evaluator.Tag("a"))!!.absUrl("href"))
```
Not sure why you're using `Evaluator.Tag`. Performance?

Reply:
Yeah, if there is no reason to use a plain css selector, I think using the Evaluator is better. I suppose this is the case, tbh, I didn't check the sources.